pkgs/os-specific/linux/dcgm/default.nix

```diff
@@ -123,7 +124,7 @@ in gcc11Stdenv.mkDerivation rec {
     fmt_9
     jsoncpp-static
     libevent-nossl-static
-    yaml-cpp
+    yaml-cpp-static
```

Comment:
[I don't mind, but] why force static and not conditional

Reply:
Do you mean for `yaml-cpp` specifically or for all dependencies? Some of them must be static for DCGM's build system to work (unless we patch it). However, DCGM's build is optimized for docker-based or RPM/Deb packaging, which doesn't have the same guarantees that Nix provides. Ideally, we'd probably not use static for any dependencies.

That said, I'm not interested in maintaining patches to DCGM's CMake build system and it's closure is already enormous, so it's not a meaningful cost. If we are able to get away with dynamic dependencies in some cases, we'll still have to override them to use `gcc11Stdenv`.

Reply:
I was only looking at `yaml-cpp` since that was in the diff

Reply:
`yaml-cpp.override { stdenv = gcc11Stdenv; }` also fixes the build, do you prefer it? My point above is that if we do this, we might as well see if we can do it to other dependencies as well. Upstream uses static linking for everything.

Reply:
I think it's best that we choose between static/dynamic linkage based on `hostPlatform.isStatic`. You can rewrite the other dependencies accordingly too if you feel up to it. It would be nice if you did, but it's not necessary for this PR

Reply:
I'll see what I can do.

Reply:
I pushed some changes to parameterize static-ness on DCGM. It's a bit silly at the moment since I've confirmed that DCGM's build only allows yaml-cpp to *actually* be linked dynamically. So setting `static = false` really only affects `yaml-cpp`, but the machinery is in place that allows us to improve it over time with fewer changes.